packages/qwik/src/lib/context.tsx

```diff
@@ -0,0 +1,23 @@
+import { 
+  createContextId, 
+  useContext
+} from "@builder.io/qwik";
+import type { SsgoiContext, SsgoiConfig } from "./types";
+
+export const SsgoiContextId = createContextId<{ 
+  context?: SsgoiContext,
+  config?: SsgoiConfig 
+}>("ssgoi-context");
+
+export const useSsgoi = () => {
+  const contextStore = useContext(SsgoiContextId);
+  const context = contextStore.context;
+  
+  if (!context) {
+    throw new Error("useSsgoi must be used within a Ssgoi component");
+  }
+  
+  return context;
+};
+
+export const SsgoiProvider = SsgoiContextId;
```

Comment:
why do you re-export the id? You have to put `useContextProvider(SsgoiContextId, {context, config})` near the top of your app, perhaps in root.tsx, so instead export

```ts
export const useSsgoiProvider = () => {
  // get context and config somehow
  useContextProvider(SsgoiContextId, {context, config})
}
```